src/main/kotlin/org/jitsi/rtp/rtcp/rtcpfb/transport_layer_fb/tcc/LastChunk.kt

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright @ 2018 - present 8x8, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jitsi.rtp.rtcp.rtcpfb.transport_layer_fb.tcc
+
+import kotlin.math.min
+
+// NOTE(brian): We have to store Chunk as an Int to avoid sign issues
+typealias Chunk = Int
+
+/**
+ * This class is a port of TransportFeedback::LastChunk in
+ * transport_feedback.h/transport_feedback.cc in Chrome
+ * https://cs.chromium.org/chromium/src/third_party/webrtc/modules/rtp_rtcp/source/rtcp_packet/transport_feedback.h?l=95&rcl=20393ee9b7ba622f254908646a9c31bf87349fc7
+ *
+ * Because of this, it explicitly does NOT try to conform
+ * to Kotlin style or idioms, instead striving to match the
+ * Chrome code as closely as possible in an effort to make
+ * future updates easier.
+ */
+class LastChunk {
+    fun Empty(): Boolean {
+        return size_ == 0
+    }
+
+    fun Clear() {
+        size_ = 0
+        all_same_ = true
+        has_large_delta_ = false
+    }
+
+    // Return if delta sizes still can be encoded into single chunk with added
+    // |delta_size|.
+    fun CanAdd(deltaSize: DeltaSize): Boolean {
+        if (size_ < kMaxTwoBitCapacity)
+            return true
+        if (size_ < kMaxOneBitCapacity && !has_large_delta_ && deltaSize != kLarge)
+            return true
+        if (size_ < kMaxRunLengthCapacity && all_same_ && delta_sizes_[0] == deltaSize)
+            return true
+        return false
+    }
+
+    // Add |delta_size|, assumes |CanAdd(delta_size)|,
+    fun Add(deltaSize: DeltaSize) {
+        if (size_ < kMaxVectorCapacity)
+            delta_sizes_[size_] = deltaSize
+        size_++
+        all_same_ = all_same_ && deltaSize == delta_sizes_[0]
+        has_large_delta_ = has_large_delta_ || deltaSize == kLarge
+    }
+
+    // Encode chunk as large as possible removing encoded delta sizes.
+    // Assume CanAdd() == false for some valid delta_size.
+    fun Emit(): Chunk {
+        if (all_same_) {
+            val chunk = EncodeRunLength()
+            Clear()
+            return chunk
+        }
+        if (size_ == kMaxOneBitCapacity) {
```

Comment:
I'm not sure if I commented about this on the old PR or somewhere. What happens if we add 13 1-byte symbols to the chunk (so we have `has_large_delta=false`, `all_same=false`) and then we try to add a large delta? CanAdd will fail, but the above check will also fail (because we have 13, not 14). Is this a problem?

Reply:
i thought this `==` was odd, too, but it's in chrome.

In this case we'll fall through here encode the first 7 statuses into a 2-bit SV chunk.